### ISSUE TITLE
Intermittent failures at endpoints lead to permanent absence

### ISSUE THREAD

**Issue by dietervu** [clarin-eric/fcs-sru-aggregator#20]
There is some evidence that a temporary failure at an endpoint leads to permanent delisting in the aggregator (the error is correctly reported in /rest/statistics). A restart of the aggregator fixes this (the endpoints show up again), but one would expect that this happens automatically upon a regular refresh of all endpoints.

**Comment by Querela** [comment]
Found this issue, seems like the periodic task fails and no refresh happens. Saw this in testing but did not think it would occur in normal running, too. But statistics page also shows an old "Start date" which should be the last scab date (not the initial one).

```java
INFO  [2023-03-01 01:26:32,752] eu.clarin.sru.fcs.aggregator.scan.ScanCrawlTask: ScanCrawlTask: Initiating crawl
ERROR [2023-03-01 01:26:32,759] eu.clarin.sru.fcs.aggregator.scan.ScanCrawlTask: !!! Scan Crawler task throwable exception
! java.lang.IllegalArgumentException: A metric named org.apache.http.conn.HttpClientConnectionManager.eu.clarin.sru.fcs.aggregator.scan.ClientFactory.available-connections already exists
! at com.codahale.metrics.MetricRegistry.register(MetricRegistry.java:168)
! at com.codahale.metrics.MetricRegistry.registerGauge(MetricRegistry.java:88)
! at com.codahale.metrics.httpclient.InstrumentedHttpClientConnectionManager.<init>(InstrumentedHttpClientConnectionManager.java:107)
! at com.codahale.metrics.httpclient.InstrumentedHttpClientConnectionManager$Builder.build(InstrumentedHttpClientConnectionManager.java:196)
! at io.dropwizard.client.HttpClientBuilder.createConnectionManager(HttpClientBuilder.java:494)
! at io.dropwizard.client.HttpClientBuilder.buildWithDefaultRequestConfiguration(HttpClientBuilder.java:303)
! at io.dropwizard.client.JerseyClientBuilder.buildConfig(JerseyClientBuilder.java:415)
! at io.dropwizard.client.JerseyClientBuilder.build(JerseyClientBuilder.java:371)
! at io.dropwizard.client.JerseyClientBuilder.build(JerseyClientBuilder.java:361)
! at eu.clarin.sru.fcs.aggregator.scan.ClientFactory.create(ClientFactory.java:60)
! at eu.clarin.sru.fcs.aggregator.scan.CenterRegistryLive.<init>(CenterRegistryLive.java:35)
! at eu.clarin.sru.fcs.aggregator.scan.ScanCrawlTask.run(ScanCrawlTask.java:73)
! at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
! at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
! at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
! at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
! at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
! at java.base/java.lang.Thread.run(Thread.java:829)
```

See also https://github.com/dropwizard/dropwizard/issues/1457, https://github.com/dropwizard/dropwizard/issues/832.
In testing is do remove all metrics first (as suggested by the dropwizard issue above):
https://github.com/clarin-eric/fcs-sru-aggregator/blob/58f6be98f365ec73e724c97650772e0643a81ae6/src/test/java/eu/clarin/sru/fcs/aggregator/app/ScanCrawlerTest.java#L48-L54

Correct usage seems to be more like: https://www.dropwizard.io/en/latest/manual/client.html (as final global instance that is reused)

**Comment by Querela** [comment]
For tests now used like this: https://github.com/jshort/coner/commit/4f6a622543548211dc2569f62b00dbc7c04e2f64
In normal application usage using a global instance. Need to check whether memory increases over time but [dropwizard docs](https://www.dropwizard.io/en/latest/manual/client.html) would suggest this is the intended usage.

Might now be fixed in #51 .

**Comment by Querela** [comment]
So, not sure about the original issue but the periodic scans should now work again. See timestamps at https://fcs.clarin-dev.eu/stats which should be within at most a day.
